src/senaite/ast/browser/panel.py

```diff
@@ -136,15 +140,49 @@ def update_analyses(self, microorganism, antibiotics):
                                           antibiotics)
 
         elif not antibiotics:
-            # Remove analyses
+            # Remove analyses that can be deleted
+            analyses = filter(self.can_delete, analyses)
             analyses_ids = map(api.get_id, analyses)
             map(self.context._delObject, analyses_ids)  # noqa
```

Comment:
Note that you are iterating here three times over the analyses.
In such a case I would rather go for a `for analysis in analyses` loop and do everything in there.